### PR TITLE
feat: add a correspondent component

### PR DESCRIPTION
**Pull request by kshychko**
## Intent of Change

- New feature (non-breaking change which adds functionality)

## Description

Adds the definition for a correspondent component such as AWS Pinpoint
## Motivation and Context

This allows creating a component for marketing communication such as push, email or SMS notifications
## How Has This Been Tested?

Schema generation testing
### Prerequisite PRs:

- None

### Dependent PRs:

- hamlet-io/engine-plugin-aws#391

### Consumer Actions:

- None

### CODE REVIEW

File: providers/shared/components/correspondent/id.ftl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                "Value" : "An outbound and inbound marketing communications service such as AWS Pinpoint"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Profiles",
+                "Children" : [
+                    {
+                        "Names" : "Logging",
```
Comment (roleyfoley) [inline providers/shared/components/correspondent/id.ftl:24]: Doesn't look this is used at the moment, might be worth removing the profile configuration for now

File: providers/shared/components/firewall/id.ftl [hunk 2]
```diff
@@ -75,7 +75,7 @@
                         ]
                     }
                 ]
-            }
+            },
```
Comment (roleyfoley) [inline providers/shared/components/firewall/id.ftl:78]: This might have been fixed? maybe need to rebase your PR?

**Review by roleyfoley — CHANGES_REQUESTED** [review]

File: providers/shared/components/firewall/id.ftl (hunk 2 above)
Comment (kshychko) [reply providers/shared/components/firewall/id.ftl:78]: The comma is still missing on master

File: providers/shared/components/correspondent/id.ftl (hunk 1 above)
Comment (kshychko) [reply providers/shared/components/correspondent/id.ftl]: Profiles attribute has been removed.

**Review by roleyfoley — APPROVED** [review]

**Pull request merged**